src/index.js

```diff
@@ -1,12 +1,22 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import ReactDOM from "react-dom";
 import { BrowserRouter } from "react-router-dom";
 import App from "./App";
 import "./theme/reset.scss";
+import { IntlProvider } from "react-intl";
+import messages from "./Languages";
+const AppIndex = () => {
+  const [lang, setLang] = useState("ar");
+  useEffect(() => {
+    setLang(lang);
+  }, [lang]);
 
-ReactDOM.render(
-  <BrowserRouter>
-    <App />
-  </BrowserRouter>,
-  document.getElementById("root")
-);
+  return (
+    <IntlProvider locale={lang} messages={messages[lang]}>
+      <BrowserRouter>
+        <App lang={setLang} />
```

Comment:
change lang here to setLang, then you will need to change the rest of the names downwards